src/main/java/com/tabnine/notifications/ConnectionLostNotificationHandler.kt

```diff
@@ -0,0 +1,61 @@
+package com.tabnine.notifications
+
+import com.intellij.notification.Notification
+import com.intellij.notification.NotificationDisplayType
+import com.intellij.notification.NotificationGroup
+import com.intellij.notification.NotificationType
+import com.intellij.notification.Notifications
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.ApplicationManager
+import com.tabnine.binary.requests.config.CloudConnectionHealthStatus
+import com.tabnine.binary.requests.config.StateResponse
+import com.tabnine.general.StaticConfig
+import com.tabnine.general.Utils.getHoursDiff
+import com.tabnine.lifecycle.BinaryStateChangeNotifier
+import java.util.Date
+import java.util.concurrent.atomic.AtomicBoolean
+
+private const val INTERVAL_BETWEEN_NOTIFICATIONS_HOURS = 5
+private const val NOTIFICATION_CONTENT =
+    "<b>Tabnine lost internet connection.</b>" +
+        "<br/>" +
+        "If your internet is connected and you're seeing this message, contact Tabnine support."
+
+class ConnectionLostNotificationHandler {
+    private val ourGroup = NotificationGroup(StaticConfig.BRAND_NAME, NotificationDisplayType.STICKY_BALLOON, false)
+    private var lastNotificationTime: Date? = null
+    private var isRegistered = AtomicBoolean(false)
+
+    fun handleConnectionLostEvent() {
+        if (!isRegistered.getAndSet(true)) {
```

Comment:
you know me... invert condition 🙂 
im not commenting this on every `if`, but here you've got like 4 more indents down the road, please spare as one lol

Reply:
totally forgot you hate it 😂